content/projects/gsoc/2021/projects/git-credentials-binding.adoc

```diff
@@ -50,35 +50,90 @@ They will be able to use the git command line options of their choice, including
+system which is being used to perform git operation that requires authentication.
+Both the bindings works on behalf of the user to provide git authentication support, without their interaction with the command-line.
+
+The bindings only provide authentication support for the `Git CLI` implementation.
+Other Git implementations inside the `git client` plugin such as `JGit` and `JGit with Apache HTTP Client` are not supported.
+To facilitate this limitation users are provided with a dropbox which lists all the Git-Tool implementation of
+CLI-Git configured in Jenkins `Global System Configuration`.
+
+
+NOTE:: If no CLi-Git implemented Git-Tool is available in `Global System Configuration` then by default JGit
+implemented Git-Tool will be used.
 
 ==== Git Username and Password Binding
 
 The `gitUsernamePassword` implementation uses the Jenkins username and password
 values retrieved through Credential API, to access a remote repository  over *HTTP* protocol.
-This binding uses the `GIT_ASKPASS` environment variable to provide credentials requested by command line git.
-The program specified by this variable is invoked with a suitable prompt on the command-line, and the user’s credential are read from its standard output.
+The binding uses the `GIT_ASKPASS` environment variable to provide credentials requested by a git operation in a pipeline job/freestyle project.
+Executable script(sh,bat,powershell) is attached/binded to `GIT_ASKPASS` variable, which is invoked when asked for user’s credentials by HTTP server.
 
-image::/images/gsoc/2021/git-credentials/GitUsernamePassword.png[Git Username and Password]
+Pipeline Job:: Using Git Username and Password Binding in a Pipeline Job
 
-IMPORTANT: By default, two variable bindings `GIT_USERNAME` and `GIT_PASSWORD`,
-bind with user's credential i.e. username and password respectively. These variable bindings are available in both freestyle project and pipeline script.
+image::../../../../images/gsoc/2021/git-credentials/usernamepassword-binding-pipline-job.png[Git Username and Password in Pipeline Job]
 
-The `gitUsernamePassword` binding only provides authentication support for the `Git CLI` implementation.
-Other Git implementations inside the `git client` plugin such as `JGit` and `JGit with Apache HTTP Client` are not supported.
+Pipeline Job:: Using Git Username and Password Binding in a FreeStyle-Project
+
+image::../../../../images/gsoc/2021/git-credentials/usernamepassword-binding-freestyle-project.png[Git Username and Password in FreeStyle Project]
+
+IMPORTANT: By default, two variable bindings `GIT_USERNAME` and `GIT_PASSWORD`,
+bind with user's credential i.e. username and password respectively.
+These variable bindings are also accessible when using `JGit/JGit with Apache HTTP Client` Git-Tool
+implementations in both freestyle project and pipeline job.
```

Comment:
```suggestion
implementations in both freestyle jobs and pipeline jobs.
```

Reply:
Is it Freestyle project or Freestyle job, I have always used Freestyle project.

Reply:
The words "job" and "project" are synonyms in Jenkins.  My issue was having a single sentence that uses both words.  I feared that someone might not realize that "job" and "project" are synonyms in Jenkins.  Just checked the [glossary](https://www.jenkins.io/doc/book/glossary/) and it says that "job" is a deprecated synonym for "project".  So, my change is in the wrong direction.  It should have changed "job" to "project" rather than changing "project" to "job"